lib/api/core/models/repositories/repository.js

```diff
@@ -0,0 +1,258 @@
+var
+  _ = require('lodash'),
+  q = require('q'),
+  RequestObject = require('../requestObject'),
+  ResponseObject = require('../responseObject');
+
+function Repository (kuzzle, options) {
+  this.collection = options.collection;
+  this.ttl = options.ttl || kuzzle.config.repositories.cacheTTL;
+  this.ObjectConstructor = options.ObjectConstructor;
+
+  this.readEngine = options.readEngine || kuzzle.services.list.readEngine;
+  this.writeEngine = options.writeEngine || kuzzle.services.list.writeEngine;
+  this.cacheEngine = options.cacheEngine || kuzzle.services.list.internalCache;
+}
+
+/**
+ *
+ * @param {string} id
+ * @returns {Promise} resolves on a new ObjectConstructor()
+ */
+Repository.prototype.loadOneFromDatabase = function (id) {
+  var
+    deferred = q.defer(),
+    requestObject,
+    result;
+
+  requestObject = new RequestObject({
+    controller: 'read',
+    action: 'get',
+    requestId: 'foo',
```

Comment:
should be removed to let the RequestObject constructor generate a random requestId